plugins/pencilblue/controllers/api/content/article_api_controller.js

```diff
@@ -41,26 +42,52 @@ module.exports = function(pb) {
     ArticleApiController.prototype.init = function(context, cb) {
         var self = this;
         var init = function(err) {
-            
+
             /**
-             * 
+             *
              * @property service
              * @type {ArticleServiceV2}
              */
             self.service = new ArticleServiceV2(self.getServiceContext());
-            
+
             /**
              *
              * @property commentService
              * @type {CommentService}
              */
             self.commentService = new CommentService(self.getServiceContext());
-                
+
             cb(err, true);
         };
         ArticleApiController.super_.prototype.init.apply(this, [context, init]);
     };
-    
+
+    /**
+     * Retrieves one or more resources from a collection.
+     * @method getAll
+     * @param {Function} cb
+     */
+    ArticleApiController.prototype.getAll = function(cb) {
+        var self = this;
+        var options = this.processQuery();
+
+        this.service.getAllWithCount(options, function(err, obj) {
+            if (util.isError(err)) {
+                return cb(err);
+            }
+            else if (util.isNullOrUndefined(obj)) {
+                return self.notFound(cb);
+            }
+
+            var userService = new UserService(self.getServiceContext());
+            userService.getAuthors(obj.data, function(err, articlesWithAuthorNames) {
```

Comment:
There is an option that can be passed to the ```ArticleServiceV2.getAll``` function that will trigger this to happen.  I _think_ all you will have to do in the article api controller is:

```
ArticleApiController.prototype.processQuery = function() {
  var options = ArticleApiController.super_.prototype.processQuery.apply(this);
  options.render = !!this.query.render; //pass 1 for true, 0 or nothing for false
  return options;
};
```